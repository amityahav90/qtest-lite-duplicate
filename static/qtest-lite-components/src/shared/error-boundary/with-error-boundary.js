import React from 'react';
import { LogContext } from '../log-service';
import { ErrorBoundary } from './error-boundary.component';

export function withErrorBoundary(WrappedComponent, fallbackRender) {
	function wrapped(props) {
		return (
			<LogContext.Consumer>
				{(value) => (
					<ErrorBoundary logPath={value} fallbackRender={fallbackRender}>
						<WrappedComponent {...props} />
					</ErrorBoundary>
				)}
			</LogContext.Consumer>
		);
	}

	return wrapped;
}
