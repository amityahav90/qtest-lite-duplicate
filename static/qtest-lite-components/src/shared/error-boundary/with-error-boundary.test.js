import React from 'react';
import renderer from 'react-test-renderer';
import { withErrorBoundary } from './with-error-boundary';

describe('Test "withErrorBoundary" wrapper component', () => {
	beforeAll(() => {
		jest.mock('@forge/bridge');
	});

	test('Component rendered correctly', async () => {
		function TestComponent() {
			return <div>test</div>;
		}

		const TestComponentWithEB = withErrorBoundary(TestComponent);
		const component = renderer.create(<TestComponentWithEB />);

		expect(component.toJSON()).toMatchSnapshot();
	});

	test('"fallbackRender" functionality rendered correctly', async () => {
		function ThrowError() {
			throw new Error('fallbackRender functionality');
		}

		function FallbackRender() {
			return <div>fallbackRender functionality worked</div>;
		}

		const TestComponentWithEB = withErrorBoundary(ThrowError, () => <FallbackRender />);
		const component = renderer.create(<TestComponentWithEB />);

		expect(component.toJSON()).toMatchSnapshot();
	});
});
