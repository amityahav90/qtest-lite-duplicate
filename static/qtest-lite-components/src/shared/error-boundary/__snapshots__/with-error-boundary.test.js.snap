// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test "withErrorBoundary" wrapper component "fallbackRender" functionality rendered correctly 1`] = `
<div>
  fallbackRender functionality worked
</div>
`;

exports[`Test "withErrorBoundary" wrapper component Component rendered correctly 1`] = `
<div>
  test
</div>
`;
